app/lib/service/openid/jwt.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+/// Pattern for a valid JWT, these must 3 base64 segments separated by dots.
+final _jwtPattern = RegExp(
+    r'^[a-zA-Z0-9+/=_-]{4,}\.[a-zA-Z0-9+/=_-]{4,}\.[a-zA-Z0-9+/=_-]{4,}$');
+
+/// Returns true if the format of the [token] could be a JWT based on the pattern.
+bool looksLikeJWT(String token) {
+  return _jwtPattern.hasMatch(token);
+}
+
+/// The decoded JWT token.
+class JWT {
```

Comment:
nit..
```suggestion
/// Representation of a JWT as defined by [RFC7519][1].
///
/// [1]: https://datatracker.ietf.org/doc/html/rfc7519
class JsonWebToken {
```
having a reference to the RFC is really nice.